dask/optimization.py

```diff
@@ -938,6 +938,42 @@ def key_split(s):
         return "Other"
 
 
+def callable_from_subgraph(dsk, outkey, inkeys, name=None):
+    """Create a callable object from a dask graph.
+
+    Parameters
+    ----------
+    dsk : dict
+        A dask graph
+    outkey : hashable
+        The output key from the graph
+    inkeys : list
+        A list of keys to be used as arguments to the callable.
+    name : str, optional
+        The name to use for the function.
+
+    Returns
+    -------
+    callable :
+        - SubgraphCallable if ``len(dsk) != 1``
+        - ``dsk[outkey][0]`` if ``len(dsk) == 1 and dsk[outkey][1:] == inkeys``
+        - TaskCallable if ``len(dsk) == 1 and dsk[outkey][1:] != inkeys``
+    """
+    if type(inkeys) is not tuple:
+        inkeys = tuple(inkeys)
+    if len(dsk) == 1:
+        task = dsk[outkey]
+        try:
+            if task[1:] == inkeys:
+                # If the task is like `(add, '_0', '_1')`, then return `add`
+                return task[0]
+        except (TypeError, ValueError):  # not observed, but just in case
+            pass
+        return TaskCallable(task, inkeys, name=name)
```

Comment:
What's a situation where this might occur? I don't understand why we can't skip wrapping a task in another object when there's only one task in the subgraph.

Reply:
We are normalizing the call signature to accept `inkeys` as args.  For example,
```python
>>> def f(*args):
...     return args
...
>>> dsk = {'a': (f, 'b', 'b')}
>>> new_callable = dask.optimization.callable_from_subgraph(dsk, 'a', ['b'])
>>> new_callable(1)
(1, 1)
>>> type(new_callable)
<class 'dask.optimization.TaskCallable'>
```
Even if there is a single task in `dsk`, the task could be very complicated, and it might not match the call signature that we want.

Reply:
I can look into maybe updating `make_blockwise_graph` to not use `TaskCallable` (or `SubgraphCallable`).  We could probably use `subs`.  The may affect our ability to perform substitution in `dask.array.core` though... dunno.

Reply:
No, I think what you have here is fine. `subs` has it's own set of issues (slow, need to traverse all tasks).